In [12]:
import pandas as pd
df=pd.read_html("https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M")

checking the number of list in the pandas to get our perticular dataframe from the list 

In [13]:
len(df)

3

In [77]:
df[0]

Postcode           Borough             Neighbourhood
2        M3A        North York                 Parkwoods
3        M4A        North York          Victoria Village
4        M5A  Downtown Toronto              Harbourfront
5        M6A        North York          Lawrence Heights
6        M6A        North York            Lawrence Manor
..       ...               ...                       ...
281      M8Z         Etobicoke  Kingsway Park South West
282      M8Z         Etobicoke                 Mimico NW
283      M8Z         Etobicoke        The Queensway West
284      M8Z         Etobicoke     Royal York South West
285      M8Z         Etobicoke            South of Bloor

[210 rows x 3 columns]

There are three list which represent the dataframe, i'm going to display the first index of the list which is our 
requird data

In [14]:
df[0]

Postcode           Borough          Neighbourhood
0        M1A      Not assigned           Not assigned
1        M2A      Not assigned           Not assigned
2        M3A        North York              Parkwoods
3        M4A        North York       Victoria Village
4        M5A  Downtown Toronto           Harbourfront
..       ...               ...                    ...
282      M8Z         Etobicoke              Mimico NW
283      M8Z         Etobicoke     The Queensway West
284      M8Z         Etobicoke  Royal York South West
285      M8Z         Etobicoke         South of Bloor
286      M9Z      Not assigned           Not assigned

[287 rows x 3 columns]

we need to store our dataframe in some variable so we store our data in the dataframe variable 

In [15]:
dataframe=df[0]

In [79]:
dataframe.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 210 entries, 2 to 285
Data columns (total 3 columns):
Postcode         210 non-null object
Borough          210 non-null object
Neighbourhood    210 non-null object
dtypes: object(3)
memory usage: 16.6+ KB


Our primary task is droping the unwanted data from the table, i have created variable df_filter to store the unwanted
data from table and use drop function to drop the data from our dataframe

In [17]:
df_filter=dataframe[(dataframe['Borough']=='Not assigned') &(dataframe['Neighbourhood']=='Not assigned')].index

In [18]:
 dataframe.drop(df_filter, inplace=True)

In [80]:
dataframe

Postcode           Borough             Neighbourhood
2        M3A        North York                 Parkwoods
3        M4A        North York          Victoria Village
4        M5A  Downtown Toronto              Harbourfront
5        M6A        North York          Lawrence Heights
6        M6A        North York            Lawrence Manor
..       ...               ...                       ...
281      M8Z         Etobicoke  Kingsway Park South West
282      M8Z         Etobicoke                 Mimico NW
283      M8Z         Etobicoke        The Queensway West
284      M8Z         Etobicoke     Royal York South West
285      M8Z         Etobicoke            South of Bloor

[210 rows x 3 columns]

we can see our clean data above in our dataframe

In [49]:
dataframe.shape

(210, 3)

There are 210 rows and 3 columns available which we need for our analysis purpose.

importing another dataframe which represent the logitude and latitude of the postal code
and we going to merge the dataframe based on postal code as a common field of both table

In [8]:
import pandas as pd
df1= pd.read_csv("Geospatial_Coordinates.csv")

In [9]:
df1

Postal Code   Latitude  Longitude
0           M1B  43.806686 -79.194353
1           M1C  43.784535 -79.160497
2           M1E  43.763573 -79.188711
3           M1G  43.770992 -79.216917
4           M1H  43.773136 -79.239476
..          ...        ...        ...
98          M9N  43.706876 -79.518188
99          M9P  43.696319 -79.532242
100         M9R  43.688905 -79.554724
101         M9V  43.739416 -79.588437
102         M9W  43.706748 -79.594054

[103 rows x 3 columns]

In [ ]:
i'm going to merge two dataframe below and store the new merge dataframe into  df4

In [73]:
df4=pd.merge(dataframe,df1, how='inner', on=['Postcode'])

In [74]:
df4

Postcode           Borough             Neighbourhood   Latitude  Longitude
0        M3A        North York                 Parkwoods  43.753259 -79.329656
1        M4A        North York          Victoria Village  43.725882 -79.315572
2        M5A  Downtown Toronto              Harbourfront  43.654260 -79.360636
3        M6A        North York          Lawrence Heights  43.718518 -79.464763
4        M6A        North York            Lawrence Manor  43.718518 -79.464763
..       ...               ...                       ...        ...        ...
205      M8Z         Etobicoke  Kingsway Park South West  43.628841 -79.520999
206      M8Z         Etobicoke                 Mimico NW  43.628841 -79.520999
207      M8Z         Etobicoke        The Queensway West  43.628841 -79.520999
208      M8Z         Etobicoke     Royal York South West  43.628841 -79.520999
209      M8Z         Etobicoke            South of Bloor  43.628841 -79.520999

[210 rows x 5 columns]

In [85]:
shape= df5.shape

In [86]:
print("total number of columns and row are :{}".format(df5.shape))

total number of columns and row are :(210, 5)
